service/rtc/sfu.go

```diff
@@ -252,6 +254,14 @@ func (s *Server) InitSession(cfg SessionConfig, closeCb func() error) error {
 			s.log.Error("failed to create ICE message", mlog.Err(err), mlog.String("sessionID", cfg.SessionID))
 			return
 		}
+
+		select {
+		case <-us.closeCh:
```

Comment:
Yes, but we do it under lock (`us.mut`), that's the key essentially. So we are guaranteeing that either `closeCh` has been closed already when we are inside this block and we exit early otherwise `s.receiveCh` must be still open and we proceed. Unfortunately I couldn't merge the two selects because the order of selection is not deterministic so we could still trigger a race.

Reply:
Right, it's random on purpose. Gotcha thanks!